gctscript/modules/gct/common.go

```diff
@@ -0,0 +1,408 @@
+package gct
+
+import (
+	"errors"
+	"fmt"
+
+	objects "github.com/d5/tengo/v2"
+	"github.com/thrasher-corp/gocryptotrader/common/file"
+	"github.com/thrasher-corp/gocryptotrader/gctscript/modules/ta/indicators"
+)
+
+var commonModule = map[string]objects.Object{
+	"writeascsv": &objects.UserFunction{Name: "writeascsv", Value: WriteAsCSV},
+}
+
+// WriteAsCSV takes in a slice matrix to save to file
+func WriteAsCSV(args ...objects.Object) (objects.Object, error) {
+	if len(args) == 0 {
+		return nil, errors.New("cannot write to file, no data present")
+	}
+
+	var bucket [][]string
+	var err error
+	var target string
+	for i := range args {
+		if args[i] == nil {
+			return nil, errors.New("data is nil")
+		}
+		var front bool
+		var temp [][]string
+		switch args[i].TypeName() {
+		case indicators.AverageTrueRange:
+			temp, err = convertATR(args[i])
+		case indicators.BollingerBands:
+			temp, err = convertBollingerBands(args[i])
+		case indicators.ExponentialMovingAverage:
+			temp, err = convertEMA(args[i])
+		case indicators.MovingAverageConvergenceDivergence:
+			temp, err = convertMACD(args[i])
+		case indicators.MoneyFlowIndex:
+			temp, err = convertMFI(args[i])
+		case indicators.OnBalanceVolume:
+			temp, err = convertOBV(args[i])
+		case indicators.RelativeStrengthIndex:
+			temp, err = convertRSI(args[i])
+		case indicators.SimpleMovingAverage:
+			temp, err = convertSMA(args[i])
+		case indicators.OHLCV:
+			temp, err = convertOHLCV(args[i])
+			front = true
+		case "string":
```

Comment:
By removing the output/string case you can also not worry about weird validations like 
`common.writeascsv(output,ret,ret2,"C:\\lol\\*&#$^@$%@$#%@$#")` where the second string overrides the first, and plus with the above, it prevents the rest of the script from running